lib/spells/TargetCondition.cpp

```diff
@@ -120,10 +116,10 @@ class AbsoluteLevelCondition : public TargetConditionItemBase
 		cachingStr << "type_" << Bonus::LEVEL_SPELL_IMMUNITY << "addInfo_1";
 
 		TBonusListPtr levelImmunities = target->getBonuses(Selector::type(Bonus::LEVEL_SPELL_IMMUNITY).And(Selector::info(1)), cachingStr.str());
-
-		if(levelImmunities->size() > 0 && levelImmunities->totalValue() >= m->getSpellLevel() && m->getSpellLevel() > 0)
-			return false;
-		return true;
+		
+		return !(levelImmunities->size() > 0
+		&& levelImmunities->totalValue() >= m->getSpellLevel()
+		&& m->getSpellLevel() > 0);
```

Comment:
And again IMHO usage of ```!``` is confusing in situations like this one.

Reply:
Agreed with 100%

Reply:
The problem is that already something like that is in `NormalLevelCondition`, so it's code dublication and I don't want want to change one without the other. Will stick for now to be certain in this place.